공고 신청, 승인, 거절 알림 생성 로직 구현

yoo-sh-96 opened the pull request:
## ✅ PR 요약
- 관련 이슈 번호: #103
- 작업 요약: 리시버 및 서비스에서 알림생성 구현, 관련 테스트 코드 작성

## 📄 상세 내용
- [x] Application 모델에 DB저장 시 post_save를 감지하는 리시버 생성
- [x] 서비스 레이어에 공고 지원시 발생하는 알림 생성 로직 구현
- [x] 관련 테스트 코드 작성

## 📸 스크린샷 (선택)
<img width="1379" height="570" alt="스크린샷 2025-09-14 오후 6 27 35" src="https://github.com/user-attachments/assets/38622761-19c9-4a46-bb23-13226ab512b6" />

## 📝 기타 참고 사항
비동기 백그라운 테스크 및 sse 연동은 추후 구현 예정

## 🧪 PR Checklist
- [x] 커밋 메시지 컨벤션에 맞게 작성했습니다.
- [x] 변경 사항에 대한 테스트를 했습니다.(버그 수정/기능에 대한 테스트).

yoo-sh-96 committed:
✨ feat: add_application 알림 생성 로직 구현

리시버 및 서비스에서 알림생성 구현, 관련 테스트 코드 작성

관련 이슈: #103

Umdoong reviewed (dismissed):
확인했습니다.
나중에 비동기 처리에 대한 것도 고민해보시면 좋을 것 같아요

yoo-sh-96 committed:
✨ feat: 스터디 공고에 지원 승인 및 거절 알림 생성

지원 상태가 pending상태인 경우에 승인 및 거절 업데이트 시 알림 생성이기에 pre_save로 저장 직전 상태 확인, status가 accept 또는 reject 일때 각각 알림 생성 로직 구현

yoo-sh-96 committed:
🎨 style: black, isort 코드 포매터 실행

black 실행 후 재커밋

yoo-sh-96 committed:
✅ test: 공고 지원 승인, 거절 알림 테스트 추가

공고 지원 승인, 거절 알림 테스트 추가

관련 이슈: #103
